core/templates/dev/head/components/state-editor/state-responses-editor/state-responses.directive.ts

```diff
@@ -568,6 +568,47 @@ angular.module('oppia').directive('stateResponses', [
             var activeStateName = $scope.stateName;
             return outcome && (outcome.dest === activeStateName);
           };
+
+          var getDeletedChoicesIndex = function(
+              newAnswerChoices, oldAnswerChoices) {
+            oldAnswerChoicesArray = oldAnswerChoices.map(function(elem) {
+              return elem.label;
+            }) ;
+            newAnswerChoicesArray = newAnswerChoices.map(function(elem) {
+              return elem.label;
+            }) ;
+            var deletedChoicesIndex = oldAnswerChoicesArray.filter(
+              function(elem) {
+                return newAnswerChoicesArray.indexOf(elem) === -1;
+              }
+            ).map(function(elem) {
+              return oldAnswerChoicesArray.indexOf(elem) ;
+            });
+            var finalDeletedChoicesIndex = [];
+            for (var i = 0; i < deletedChoicesIndex.length; i++) {
+              finalDeletedChoicesIndex.push(deletedChoicesIndex[i] - i);
+            }
+            return finalDeletedChoicesIndex;
+          };
+
+          var validateAnwerGroupsOnChoiceDeletion = function(index) {
+            var requiresUpdation = false;
+            var answerGroups = ResponsesService.getAnswerGroups();
+            for (var i = 0; i < answerGroups.length; i++) {
+              var rules = answerGroups[i].rules;
+              for (var j = 0; j < rules.length; j++) {
+                if (index < rules[j].inputs.x) {
+                  ResponsesService.reduceRuleIndexByOne(i, j);
+                  requiresUpdation = true;
```

Comment:
You can just `return true` here and below, and return false at the end of the function.

Reply:
The way this is implemented, we need to run the second loop `rules.length` times(In the case when multiple rules are present). So if we return before the loop runs this many times it will not give the expected result.
That is why we just return a single boolean in the end

Reply:
Yeah, makes sense!